plugins/plextor/qscan_cmd_plextor.cpp

```diff
@@ -0,0 +1,795 @@
+    dest[pit][min(v, len - 1)]++;
+    idx += 2;
+  }
+  return 0;
+}
+
+int scan_plextor::build_TA_histogram_px755(unsigned char *response_data,
+                                           int *dest_pit, int *dest_land,
+                                           int len, int dt) { //, int spd) {
+  int *dest[] = {dest_land, dest_pit};
+  int count = ntoh16(response_data + 2);
+  //	printf("PX-755/760 Histogram... %d\n", count);
+  int idx = 28;
+  int v, pit;
+  for (int i = 0; i < count; i++) {
+    v = ntoh16u(response_data + idx);
+    pit = !!(v & 0x8000);
+    v &= ~0x8000;
+    if (dt & DISC_DVDplus)
+      dest[pit][min((int)(v * 1.452), len - 1)]++; // DVD+R(W), speed 2.4x
+    else
+      dest[pit][min((int)(v * 1.21), len - 1)]++; // DVD-R(W), speed 2x
+    idx += 2;
+  }
+  return 0;
+}
+
+int scan_plextor::evaluate_histogramme(cdvd_ta *data, int **peaks, int **mins) {
+  int i, j1, j2;
+  int local_max = 0;
+  int next_peak = 0;
+  int peak_found = 0;
+  int *src[] = {data->pit, data->land};
+
+  for (int k = 0; k < 2; k++) {
+    j1 = 0;
+    j2 = 0;
+    local_max = 0;
+    for (i = 40; i < 330; i++) {
+      if (src[k][i - 1] <= src[k][i] && src[k][i + 1] <= src[k][i] &&
+          src[k][i] > 20 && src[k][i] > local_max) {
+        peaks[k][j1] = i;
+        local_max = src[k][i];
+        next_peak = 1;
+      } else if (peak_found)
+        if (/*src[k][i-3] >= src[k][i-1] && src[k][i-2] > src[k][i-1] && */
+            src[k][i - 1] > src[k][i] && src[k][i + 1] >= src[k][i]) {
+          mins[k][j2] = i;
+          if (j2 < 13)
+            j2++;
+          peak_found = 0;
+        }
+
+      if (local_max > 2 * src[k][i]) {
+        local_max = 2 * src[k][i];
+        if (next_peak) {
+          next_peak = 0;
+          if (j1 < 13) {
+            j1++;
+            peak_found = 1;
+            //	printf("%4d",i);
+          }
+        }
+      }
+    }
+
+    int min_count = j2;
+    for (i = 0; i < min_count; i++) {
+      int start = (i ? mins[k][i - 1] : 0);
+      int end = mins[k][i];
+      int sum = 0;
+      int partsum = 0;
+      int j;
+      for (j = start; j < end; sum += src[k][j++])
+        ;
+      for (j = start; partsum < sum / 2; partsum += src[k][j++])
+        ;
+      peaks[k][i] = (peaks[k][i] + j - 1) / 2;
+    }
+  }
+  return 0;
+}
+
+int scan_plextor::cmd_dvd_ta_block(cdvd_ta *data) {
+  //	printf("Start PLEXTOR TA test on %d layers\n",dev->media.layers);
+  int i, r, m, j;
+  float sum;
+  unsigned char scan_cmd[6][2] = {{0x04, 0x00}, {0x10, 0x00}, {0x20, 0x00},
+                                  {0xFA, 0x28}, {0xEA, 0x28}, {0xDE, 0x28}};
+  const char *scan_txt[] = {
+      "Running TA on L0 inner zone ", "Running TA on L0 middle zone",
+      "Running TA on L0 outer zone",  "Running TA on L1 inner zone ",
+      "Running TA on L1 middle zone", "Running TA on L1 outer zone"};
+  //	int ta_response_pit[6][512]; int ta_response_land[6][512];
+  int peaks_lands[15], peaks_pits[15];
```

Comment:
Codacy found an issue: [Variable 'peaks_lands' is not assigned a value.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6072101)